### PR TITLE
Fix mac os info.plist entries

### PR DESCRIPTION
**Pull request by warp-core**
**Bug fix**

Reported by comradecandy on Discord.

## Summary
Some of the substitutions in the info.plist file used included in the MacOS package are incorrect, and so those values are not being correctly set by CMake.
One of the issues this results in is the name of the whole package needing to be the same as the executable within it. If the package is renamed, MacOS will no longer be able to find the executable within it and it will no longer run.
This PR corrects the substitution used for the executable name so it should be set correctly by CMake, as well as correcting syntax for the bundle identifier replacement and the name used for the minimum version replacement.

## Testing Done
We'll see when it builds.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -223,7 +223,8 @@ if(APPLE AND ES_CREATE_BUNDLE)
 	set_target_properties(EndlessSky PROPERTIES
 		MACOSX_BUNDLE_INFO_PLIST "${CMAKE_CURRENT_LIST_DIR}/resources/EndlessSky-Info.plist"
 		XCODE_ATTRIBUTE_ASSETCATALOG_COMPILER_APPICON_NAME endless-sky)
-	set_target_properties(EndlessSky PROPERTIES OUTPUT_NAME "Endless Sky")
+	set(OUTPUT_NAME "Endless Sky")
+	set_target_properties(EndlessSky PROPERTIES OUTPUT_NAME ${OUTPUT_NAME})
```
Comment (tibetiroka) [inline CMakeLists.txt:227]: Is this change really necessary? And if it is, shouldn't it be applied to the executables below?

Comment (warp-core) [reply CMakeLists.txt:227]: I've done this so that any changes to the output name for the Mac OS build here will be automatically propagated to the plist file. I haven't done the same for the Windows or Linux builds because, as far as I know, those names don't need to be set by CMake anywhere else the way the Mac OS build needs it in the plist file. I haven't refused this variable there because the Linux output name is actually different already and the Windows output name isn't necessarily the same as the Mac OS one, it can be different, they just happen to have the same value at the moment.
